browser/speedreader/speedreader_tab_helper.cc

```diff
@@ -5,21 +5,38 @@
 
 #include "brave/browser/speedreader/speedreader_tab_helper.h"
 
+#include <memory>
+
 #include "brave/browser/brave_browser_process_impl.h"
 #include "brave/browser/speedreader/speedreader_service_factory.h"
 #include "brave/components/speedreader/speedreader_rewriter_service.h"
 #include "brave/components/speedreader/speedreader_service.h"
 #include "brave/components/speedreader/speedreader_test_whitelist.h"
+#include "chrome/browser/dom_distiller/dom_distiller_service_factory.h"
+#include "chrome/browser/dom_distiller/tab_utils.h"
 #include "chrome/browser/profiles/profile.h"
+#include "components/dom_distiller/content/browser/distiller_page_web_contents.h"
+#include "components/dom_distiller/content/browser/uma_helper.h"
+#include "components/dom_distiller/core/url_utils.h"
 #include "content/public/browser/navigation_handle.h"
 
+using dom_distiller::UMAHelper;
+using dom_distiller::url_utils::IsDistilledPage;
+
 namespace speedreader {
 
 SpeedreaderTabHelper::~SpeedreaderTabHelper() = default;
 
 SpeedreaderTabHelper::SpeedreaderTabHelper(content::WebContents* web_contents)
     : content::WebContentsObserver(web_contents) {}
 
+bool SpeedreaderTabHelper::IsReaderEnabled() const {
+  if (speedreader_active_)
+    return true;
+  auto url = web_contents()->GetURL();
+  return IsDistilledPage(url);
```

Comment:
`GetURL()` is deprecated
`return IsDistilledPage(web_contents()->GetLastCommitedURL())`